### ISSUE TITLE
Save history

### ISSUE THREAD

**Issue by ianb** [mozilla/firefox-voice#1237]
We should save the history of everything you say to Firefox Voice.

This should happen in a new module in `extension/history.js`. `intentRunner.js` would add history items as part of the `runIntent()` function.

The history should be saved using [IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API). In other places we use `browser.storage`, but that's not very scalable for an ever-increasing list.

IndexedDB is not awesome to use, but I'm not excited about adding a library. Still, [Dexie](https://dexie.org/) looks OK, and doesn't have any dependencies from what I see.

To add a new dependency, try to install it with `npm` and then edit `build:deps` in `package.json`. The file (hopefully it's just one file) can be copied into `extension/js/vendor/` – it appears Dexie [supports ECMA modules](https://dexie.org/docs/Tutorial/Consuming-dexie-as-a-module).

After history is saved the next step will be displaying the history to the user.

**Comment by OliviaRuyinZhang** [comment]
Hi @ianb, I would like to give it a try. Could I get this one? Thanks

**Comment by saucekode** [comment]
Hello @ianb can I please work on this. Thank you

**Comment by ianb** [comment]
I'll assign to @OliviaRuyinZhang – we'll be adding more new issues today too, so keep an eye out

**Comment by OliviaRuyinZhang** [comment]
Hi @ianb, I have finished developing a framework for indexedDB. Now, we can create table(s) to store any history we would like to store on browser. I modified intentRunner.js as well to save everything users say to Firefox Voice using indexedDB. 

I did not develope a UI to display the history to users yet since you said that would be the next step. Furthermore, I would like to see if there's any feedback or suggestion about the framework or the requirenments of the UI design before I move on.
